IBAnimatable/SystemTransitionAnimator.swift

```diff
@@ -0,0 +1,97 @@
+//
+//  SystemTransitionAnimator.swift
+//  IBAnimatable
+//
+//  Created by Steven on 4/1/17.
+//  Copyright © 2017 IBAnimatable. All rights reserved.
+//
+
+import UIKit
+
+class SystemTransitionAnimator: NSObject, AnimatedTransitioning {
+
+  var transitionDuration: Duration
+  var transitionAnimationType: TransitionAnimationType
+  var reverseAnimationType: TransitionAnimationType?
+  var interactiveGestureType: InteractiveGestureType?
+  var systemTransitionType: TransitionAnimationType.SystemTransitionType
+  var systemTransitionSubtype: String?
+
+  init(systemType: TransitionAnimationType.SystemTransitionType, duration: Duration, direction: TransitionAnimationType.Direction? = nil) {
+
+    self.systemTransitionType = systemType
+    self.transitionDuration = duration
+
+    if let direction = direction {
+      self.transitionAnimationType = TransitionAnimationType(systemType: systemType, direction: direction)
```

Comment:
We are mapping a `TransitionAnimationType` to a `SystemTransitionType ` then rolling back to `TransitionAnimationType`. There's room to improvements here I think 🤔